nipraxis-spring-2022/diagnostics-Lemanics
# Adding some basics metrics with an option to have nii for the testing phase to check the metrics

Some computation for the metrics need to be checked, not quite sure they always make sense

I have also added an output_tests folder with the output for one subject so that you can get an idea of the outputs without needing to re-run the code
The files in the output_test will then have to be deleted (I will git ignore them). But for now, I think it will be quicker for you to see the output

I have also change the argument from img (nibabel image) to image_fname (filename of the nii) as it is then a lot easier to test independently function on ipython without always doing "import nibabel"

Sorry if some metric do not make sense, please tell me if some should be deleted, this is just a first draft. ;)

Can we use toolbox such as nipype for the masking or not ? I have done an easy_masking but it will not be so nice.


#TO TEST the code
cd <your_repo_path>

ipython
#test it on sub 01
image_fname =r".\data\group-00\sub-01\func\sub-01_task-taskzero_run-01_bold.nii.gz"

#by default will save all the nii
basic_stats(image_fname) #otherwise do basic_stats(image_fname, False) for no plot
coefficient_of_variation(image_fname)
dvars(image_fname)

value_mask=106.65432098765433+2*15.243432526000921
 #meanmap_median+2*sdmap_median #here for image_fname
easy_masking(image_fname,value_mask) 

#check your output_test folder
#do git ignore for output_test folder
